spark-operator/src/main/java/org/apache/spark/k8s/operator/SparkOperator.java

```diff
@@ -0,0 +1,225 @@
+import org.apache.spark.k8s.operator.metrics.source.KubernetesMetricsInterceptor;
+import org.apache.spark.k8s.operator.metrics.source.OperatorJosdkMetrics;
+import org.apache.spark.k8s.operator.probe.ProbeService;
+import org.apache.spark.k8s.operator.reconciler.SparkAppReconciler;
+import org.apache.spark.k8s.operator.reconciler.SparkReconcilerUtils;
+import org.apache.spark.k8s.operator.utils.ClassLoadingUtils;
+import org.apache.spark.k8s.operator.utils.SparkAppStatusRecorder;
+
+/**
+ * Entry point for Spark Operator. Sets up reconcilers for CustomResource and health check servers
+ */
+@Slf4j
+public class SparkOperator {
+  private final List<Operator> operators;
+  private final KubernetesClient client;
+  private final SparkAppSubmissionWorker appSubmissionWorker;
+  private final SparkAppStatusRecorder sparkAppStatusRecorder;
+  protected Set<RegisteredController<?>> registeredSparkControllers;
+  protected Set<String> watchedNamespaces;
+  private final MetricsSystem metricsSystem;
+  private final SentinelManager<SparkApplication> sparkApplicationSentinelManager;
+  private final ProbeService probeService;
+  private final MetricsService metricsService;
+  private final ExecutorService metricsResourcesSingleThreadPool;
+
+  public SparkOperator() {
+    this.metricsSystem = MetricsSystemFactory.createMetricsSystem();
+    this.client =
+        KubernetesClientFactory.buildKubernetesClient(getClientInterceptors(metricsSystem));
+    this.appSubmissionWorker = new SparkAppSubmissionWorker();
+    this.sparkAppStatusRecorder =
+        new SparkAppStatusRecorder(
+            ClassLoadingUtils.getStatusListener(
+                SparkAppStatusListener.class,
+                SparkOperatorConf.SPARK_APP_STATUS_LISTENER_CLASS_NAMES.getValue()));
+    this.registeredSparkControllers = new HashSet<>();
+    this.watchedNamespaces = SparkReconcilerUtils.getWatchedNamespaces();
+    this.sparkApplicationSentinelManager = new SentinelManager<>();
+    this.operators = new ArrayList<>();
+    this.operators.add(createSparkOperator());
+    if (SparkOperatorConf.DYNAMIC_CONFIG_ENABLED.getValue()) {
+      this.operators.add(createSparkOperatorConfMonitor());
+    }
+    this.metricsResourcesSingleThreadPool = Executors.newSingleThreadExecutor();
+    this.probeService =
+        new ProbeService(
+            operators, Collections.singletonList(sparkApplicationSentinelManager), null);
+    this.metricsService = new MetricsService(metricsSystem, metricsResourcesSingleThreadPool);
+  }
+
+  protected Operator createSparkOperator() {
+    Operator op = new Operator(this::overrideOperatorConfigs);
+    registeredSparkControllers.add(
+        op.register(
+            new SparkAppReconciler(
+                appSubmissionWorker, sparkAppStatusRecorder, sparkApplicationSentinelManager),
+            this::overrideControllerConfigs));
+    return op;
+  }
+
+  protected Operator createSparkOperatorConfMonitor() {
```

Comment:
ditto. (https://github.com/apache/spark-kubernetes-operator/pull/12/files#r1677217454 and https://github.com/apache/spark-kubernetes-operator/pull/12/files#r1677217641)